pages/films/index.tsx

```diff
@@ -0,0 +1,276 @@
+        Sort={
+          <FilmsOrderBySelector
+            handleChangeOrderBy={handleChangeOrderBy}
+            param={filmParam}
+          />
+        }
+        Direction={
+          <SortDirectionToggler
+            handleSortDirectionToggler={handleSortDirectionToggler}
+            sortDiraction={sortDiraction}
+            filmParam={filmParam}
+          />
+        }
+        resetSearchParams={resetSearchParams}
+        searchQueries={searchQueries}
+        title={"Anime"}
+      />
+
+      <div className="search__wrapper">
+        <div className="search__navbar-container">
+          <Sidebar
+            handleChangeFilmType={handleChangeFilmType}
+            handleChangeDate={handleChangeDate}
+            filmType={filmType}
+            dateRange={dateRange}
+            defaultFilmDate={defaultFilmDate}
+            resetFiltersQueries={resetFiltersQueries}
+            filtersQueries={filtersQueries}
+          />
+        </div>
+        <div className="search__container">
+          <div className="search__filmList">
+            {store.films.map((film) => {
+              return (
+                <div key={film.mal_id}>
+                  <FilmCard
+                    title={film.title}
+                    imageUrl={film.images.jpg.image_url}
+                    synopsis={film.synopsis}
+                    score={film.score}
+                    id={film.mal_id}
+                    type={film.type}
+                    year={film.year}
+                  />
+                </div>
+              )
+            })}
+          </div>
+          <div className="search__paginator">
+            <Paginator
+              totalPagesCount={
+                store.pagination.last_visible_page ?? defaultTotalItemsCount
+              }
+              currentPage={page ?? defaultCurrentPage}
+              handleChangePage={handleChangePage}
+            />
+            <div className="search__PageSizeSelector-container">
+              <PageSizeSelector
+                handleChangePageSize={handleChangePageSize}
+                perPage={perPage ?? undefined}
+              />
+            </div>
+          </div>
+        </div>
+      </div>
+    </div>
+  )
+})
+
+export async function getServerSideProps(context: NextPageContext) {
+  const store = initializeStore()
+  await store.fetchFilms({
+    ...(context.query.page ? { page: Number(context.query.page) } : {}),
+    ...(context.query.perPage ? { limit: Number(context.query.perPage) } : {}),
+    ...(context.query.q
+      ? {
+          q: context.query.q as string,
+        }
+      : {}),
+    ...(context.query.type ? { type: context.query.type as string } : {}),
+    ...(context.query.startDate
+      ? { date_range: context.query.date_range as string }
+      : {}),
+    ...(context.query.order_by
+      ? { order_by: context.query.order_by as string }
+      : {}),
+    ...(context.query.sort ? { sort: context.query.sort as string } : {}),
+  })
+
```

Comment:
заюзать функцию которую ты делал
```suggestion
  await store.fetchFilms(assignIfDefined({
    page: context.query.page,
    limit: context.query.perPage,
    q: context.query.q,
    type: context.query.type,
    date_range: context.query.date_range,
    order_by: context.query.order_by,
    sort: context.query.sort,
  }))

```
саму функцию assignIfDefined лучше написать в отдельном файле в папке src/utils